crates/sui-indexer-alt-framework/src/pipeline/processor.rs

```diff
@@ -27,6 +27,10 @@ pub trait Processor {
     /// How much concurrency to use when processing checkpoint data.
     const FANOUT: usize = 10;
 
+    /// Whether the pruner requires processed values in order to prune.
+    /// This will determine the first checkpoint to process when we start the pipeline.
+    const PRUNING_REQUIRES_PROCESSED_VALUES: bool = false;
```

Comment:
This is kind of an unfortunate consequence of this approach, but I'm coming around to it, with some small tweaks:

- This feature is unique to concurrent pipelines (as is pruning), so we should put it on the `concurrent::Handler` trait (although I see now above why you've put it on `Processor` based on where in the codebase you fetch the watermark).
- I would name it for exactly what it controls (resuming from the pruner watermark), rather than what we might use it for (back-channeling values from the processor to the pruner). How about `RESUME_FROM_PRUNER_HI` or something?

Reply:
To me "resuming from the pruner watermark" is an implementation detail that would actually be difficult for pipeline builders to understand. What they care about is the fact that they want to use processed data to do pruning, and the need to resume ingestion from pruner hi is a side-effect.